stacks/web-application.yml

```diff
@@ -0,0 +1,257 @@
+  EcrRegion:
+    Type: String
+  SecretsBase:
+    Type: String
+  # Shared ENV #################################################################
+  # App ECS #################################################################
+  EcrImageTag:
+    Type: String
+  # App ENV #################################################################
+  AppName:
+    Type: String # castle, crier, cms, feeder
+  SecretsVersion:
+    Type: String
+  CreateWorker:
+    Type:  String
+  ContainerPort:
+    Type: String
+    Default: 3000
+  ContainerMemory:
+    Type: String
+    Default: 500
+  ContainerCpu:
+    Type: String
+    Default: 128
+  HealthCheckPath:
+    Type: String
+    Default: /api/v1
+Resources:
+# Web Resources
+  ALBTargetGroup:
+    Type: "AWS::ElasticLoadBalancingV2::TargetGroup"
+    Properties:
+      HealthCheckIntervalSeconds: 60
+      UnhealthyThresholdCount: 10
+      HealthCheckPath: !Ref HealthCheckPath
+      Name: !Sub ${AppName}-${EnvironmentTypeAbbreviation}-${VPC}
+      Port: 80
+      Protocol: HTTP
+      TargetGroupAttributes:
+        - Key: deregistration_delay.timeout_seconds
+          Value: 30
+      Tags:
+        - Key: Project
+          Value: !Ref AppName
+        - Key: Environment
+          Value: !Ref EnvironmentType
+        - Key: Name
+          Value: !Sub ${AppName}-${EnvironmentType}
+      VpcId: !Ref VPC
+  # ALB Listener Rules
+  ALBHTTPHostEnvAppListenerRule:
+    Type: "AWS::ElasticLoadBalancingV2::ListenerRule"
+    Properties:
+      Actions:
+        - TargetGroupArn: !Ref ALBTargetGroup
+          Type: forward
+      Conditions:
+        - Field: host-header
+          Values:
+            - !Sub ${EnvironmentTypeAbbreviation}-${AppName}.*
+      ListenerArn: !Ref PlatformALBHTTPListenerArn
+      Priority: !Join ["", [!Ref PlatformALBListenerPriorityPrefix, 01]]
+  ALBHTTPHostAppEnvListenerRule:
+    Type: "AWS::ElasticLoadBalancingV2::ListenerRule"
+    Properties:
+      Actions:
+        - TargetGroupArn: !Ref ALBTargetGroup
+          Type: forward
+      Conditions:
+        - Field: host-header
+          Values:
+            - !Sub ${AppName}-${EnvironmentTypeAbbreviation}.*
+      ListenerArn: !Ref PlatformALBHTTPListenerArn
+      Priority: !Join ["", [!Ref PlatformALBListenerPriorityPrefix, 02]]
+  ALBHTTPHostAppListenerRule:
+    Type: "AWS::ElasticLoadBalancingV2::ListenerRule"
+    Properties:
+      Actions:
+        - TargetGroupArn: !Ref ALBTargetGroup
+          Type: forward
+      Conditions:
+        - Field: host-header
+          Values:
+            - !Sub ${AppName}.*
+      ListenerArn: !Ref PlatformALBHTTPListenerArn
+      Priority: !Join ["", [!Ref PlatformALBListenerPriorityPrefix, 03]]
+  ALBTargetGroup500Alarm:
+    Type: "AWS::CloudWatch::Alarm"
+    # Condition: CreateProductionResources
+    Properties:
+      ActionsEnabled: true
+      AlarmName: !Sub ${AppName} ALB Target Group HTTP 5XXs
+      AlarmActions:
+        - !Ref OpsErrorMessagesSnsTopicArn
```

Comment:
I wonder if this should be a parameter. Some apps 500'ing on the web container are a lot worse than others, and I'm trying to make sure we're being fairly deliberate about which alarms show up where.